visual_behavior/ophys/io/convert_level_1_to_level_2.py

```diff
@@ -124,33 +131,73 @@ def get_experiment_date(lims_data):
+#     if 'analysis_dir' in lims_data.columns:
+#         return lims_data['analysis_dir'].values[0]
+#     analysis_dir = os.path.join(cache_dir, get_analysis_folder_name(lims_data))
+#     print(analysis_dir)
+#     if not os.path.exists(analysis_dir):
+#         print('Creating a new analysis folder')
+#         os.mkdir(analysis_dir)
+#     else:
+#         print('Analysis folder exists')
+#     # Check if more than one analysis folder exists
+#     allFiles = os.listdir(cache_dir)
+#     inds = [allFiles[i].find(str(np.squeeze(lims_data.lims_id.values))) for i in range(len(allFiles))]
+#     existingFolders = np.argwhere(np.array(inds) != -1)
+#     # Get the modification times of the existing analysis folders
+#     modifTimes = [os.path.getmtime(os.path.join(cache_dir, allFiles[np.squeeze(existingFolders[i])])) for i in
+#                   range(len(existingFolders))]
+#     # Find all the old analysis folders
+#     indsOld = np.argsort(modifTimes)[:-1]
+#     oldFiles = [os.path.join(cache_dir, allFiles[np.squeeze(existingFolders[indsOld[i]])]) for i in range(len(indsOld))]
+#     # Remove old analysis folders
+#     for i in range(len(oldFiles)):
+#         print('Removing old analysis folder : %s' % oldFiles[i])
+#         shutil.rmtree(oldFiles[i])
+#     if cache_on_lims_data:
+#         lims_data.insert(loc=2, column='analysis_dir', value=analysis_dir)
+#     return analysis_dir
+
+
 def get_analysis_dir(lims_data, cache_dir=None, cache_on_lims_data=True):
     cache_dir = get_cache_dir(cache_dir=cache_dir)
     if 'analysis_dir' in lims_data.columns:
         return lims_data['analysis_dir'].values[0]
-    analysis_dir = os.path.join(cache_dir, get_analysis_folder_name(lims_data))
-    print(analysis_dir)
-    if not os.path.exists(analysis_dir):
-        print('Creating a new analysis folder')
-        os.mkdir(analysis_dir)
     else:
-        print('Analysis folder already exists!')
-    # Check if more than one analysis folder exists
-    allFiles = os.listdir(cache_dir)
-    inds = [allFiles[i].find(str(np.squeeze(lims_data.lims_id.values))) for i in range(len(allFiles))]
-    existingFolders = np.argwhere(np.array(inds) != -1)
-    # Get the modification times of the existing analysis folders
-    modifTimes = [os.path.getmtime(os.path.join(cache_dir, allFiles[np.squeeze(existingFolders[i])])) for i in range(len(existingFolders))]
-    # Find all the old analysis folders
-    indsOld = np.argsort(modifTimes)[:-1]
-    oldFiles = [os.path.join(cache_dir, allFiles[np.squeeze(existingFolders[indsOld[i]])]) for i in range(len(indsOld))]
-    # Remove old analysis folders
-    for i in range(len(oldFiles)):
-        print('Removing old analysis folder : %s' % oldFiles[i])
-        shutil.rmtree(oldFiles[i])
-    if cache_on_lims_data:
-        lims_data.insert(loc=2, column='analysis_dir', value=analysis_dir)
-    return analysis_dir
+        # look for existing analysis folders
```

Comment:
@dougollerenshaw im not sure how these changes to the `get_analysis_folder `function could affect things

Reply:
I'm also having a hard time parsing these changes. Any thoughts on how to make it less disruptive, or do you think the small group of existing users will be able to adapt?